test/integration/controller/jobset_controller_test.go

```diff
@@ -270,6 +282,122 @@ var _ = ginkgo.Describe("JobSet controller", func() {
+								completeJob(&jobList.Items[i])
+							}
+						}
+					},
+					checkJobSetCondition: util.JobSetActive,
+				},
+				{
+					// Now complete the job in the replicated job selected by the success policy
+					// and ensure the jobset completes.
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						ginkgo.By("completing all jobs from replicated-job-b")
+						for i := 0; i < len(jobList.Items); i++ {
+							if jobList.Items[i].Labels[jobset.ReplicatedJobNameKey] == "replicated-job-b" {
+								completeJob(&jobList.Items[i])
+							}
+						}
+					},
+					checkJobSetCondition: util.JobSetCompleted,
+				},
+			},
+		}),
+		ginkgo.Entry("success policy 'all' with replicated jobs specified", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					SuccessPolicy(&jobset.SuccessPolicy{
+						Operator:           jobset.OperatorAll,
+						ReplicatedJobNames: []string{"replicated-job-a"},
+					})
+			},
+			updates: []*update{
+				{
+					// Jobset has 2 replicated jobs, but only 1 is selected in the success policy.
+					// Complete all the jobs in the other replicated job and ensure the jobset is still active.
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						ginkgo.By("completing all jobs from different replicated job")
+						for i := 0; i < len(jobList.Items); i++ {
+							if jobList.Items[i].Labels[jobset.ReplicatedJobNameKey] == "replicated-job-b" {
+								completeJob(&jobList.Items[i])
+								break
+							}
+						}
+					},
+					checkJobSetCondition: util.JobSetActive,
+				},
+				{
+					// Now complete the job in the replicated job selected by the success policy
+					// and ensure the jobset completes.
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						ginkgo.By("completing all jobs in replicated job selected by success policy")
+						for i := 0; i < len(jobList.Items); i++ {
+							if jobList.Items[i].Labels[jobset.ReplicatedJobNameKey] == "replicated-job-a" {
+								completeJob(&jobList.Items[i])
+							}
+						}
+					},
+					checkJobSetCondition: util.JobSetCompleted,
+				},
+			},
+		}),
+		ginkgo.Entry("success policy 'any' with replicated job specified", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					// If any of the 3 jobs in replicated-job-b succeeds, the jobset is marked completed.
+					SuccessPolicy(&jobset.SuccessPolicy{
+						Operator:           jobset.OperatorAny,
+						ReplicatedJobNames: []string{"replicated-job-b"},
+					})
+			},
+			updates: []*update{
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						ginkgo.By("completing 1 of 3 jobs in replicated-job-b")
+						for i := 0; i < len(jobList.Items); i++ {
+							if jobList.Items[i].Labels[jobset.ReplicatedJobNameKey] == "replicated-job-b" {
+								completeJob(&jobList.Items[i])
+								break
+							}
+						}
```

Comment:
create a function that accepts a list of jobs and returns a slice of jobs matching the wanted replicatedJob name

Reply:
done